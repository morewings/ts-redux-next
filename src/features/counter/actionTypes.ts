export enum Actions {
    INCREMENT_COUNTER = 'INCREMENT_COUNTER',
}
