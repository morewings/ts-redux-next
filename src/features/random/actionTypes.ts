export const GET_RANDOM_NUMBER = 'GET_RANDOM_NUMBER';

export enum Actions {
    GET_RANDOM_NUMBER = 'GET_RANDOM_NUMBER',
    GET_RANDOM_NUMBER_REJECTED = 'GET_RANDOM_NUMBER_REJECTED',
    GET_RANDOM_NUMBER_FULFILLED = 'GET_RANDOM_NUMBER_FULFILLED',
    GET_RANDOM_NUMBER_PENDING = 'GET_RANDOM_NUMBER_PENDING',
}
