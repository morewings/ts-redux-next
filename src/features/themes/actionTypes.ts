export const TOGGLE_THEME = 'TOGGLE_THEME';

export type Action = {
    type: string;
};
