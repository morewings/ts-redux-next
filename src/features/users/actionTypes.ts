import type { AxiosResponse } from 'axios';

export const GET_USERS = 'GET_USERS';
export const GET_INFO = 'GET_INFO';

export type Action = {
    type: string;
    payload: AxiosResponse;
};
