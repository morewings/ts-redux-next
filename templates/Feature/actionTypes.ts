export enum Actions {
    GET_TEMPLATE_NAME = 'GET_TEMPLATE_NAME',
    GET_TEMPLATE_NAME_REJECTED = 'GET_TEMPLATE_NAME_REJECTED',
    GET_TEMPLATE_NAME_FULFILLED = 'GET_TEMPLATE_NAME_FULFILLED',
    GET_TEMPLATE_NAME_PENDING = 'GET_TEMPLATE_NAME_PENDING',
}
