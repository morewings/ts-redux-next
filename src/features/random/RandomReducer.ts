import {Actions} from './actionTypes';

export type State = {
    number?: number;
    isLoading?: boolean;
    hasError?: boolean;
    isFulfilled?: boolean;
};

const initialState = {
    number: undefined,
    isLoading: false,
    hasError: false,
    isFulfilled: false,
} as State;

export type Action = {
    type: keyof typeof Actions;
    payload?: {data?: number};
};

const reducer = (state = initialState, action: Action): State => {
    switch (action.type) {
        case `${Actions.GET_RANDOM_NUMBER}_PENDING`:
            return {
                ...state,
                isFulfilled: false,
                isLoading: true,
                hasError: false,
                number: undefined,
            };

        case `${Actions.GET_RANDOM_NUMBER}_FULFILLED`:
            return {
                isFulfilled: true,
                isLoading: false,
                hasError: false,
                number: action?.payload?.data,
            };

        case `${Actions.GET_RANDOM_NUMBER}_REJECTED`:
            return {
                isFulfilled: false,
                isLoading: false,
                hasError: true,
                number: undefined,
            };

        default:
            return state;
    }
};

export default reducer;
