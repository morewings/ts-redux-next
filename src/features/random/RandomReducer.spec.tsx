import {Actions} from './actionTypes';
import RandomReducer from './RandomReducer';

describe('features > random > RandomReducer', () => {
    it('returns initial state, if non matched action is dispatched', () => {
        const initialState = {
            isLoading: false,
            hasError: false,
            isFulfilled: false,
        };

        const action = {
            type: Actions.GET_RANDOM_NUMBER,
            payload: 0,
        };

        expect(RandomReducer(initialState, action)).toBe(initialState);
    });

    /**
     * Provide table of values to run test case against
     * @see https://jestjs.io/docs/en/api#testeachtablename-fn-timeout
     */
    it.each([
        [`${Actions.GET_RANDOM_NUMBER}_FULFILLED`],
        [`${Actions.GET_RANDOM_NUMBER}_PENDING`],
        [`${Actions.GET_RANDOM_NUMBER}_REJECTED`],
    ])(`updates state according to dispatched action`, actionType => {
        const initialState = {
            number: 0,
        };

        const payload =
            actionType === `${Actions.GET_RANDOM_NUMBER}_FULFILLED`
                ? 1
                : undefined;

        const action = {
            type: actionType as keyof typeof Actions,
            payload,
        };

        expect(RandomReducer(initialState, action)).toMatchSnapshot();
    });
});
