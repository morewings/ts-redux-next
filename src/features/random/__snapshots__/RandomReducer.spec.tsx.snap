// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`features > random > RandomReducer updates state according to dispatched action 1`] = `
{
  "hasError": false,
  "isFulfilled": true,
  "isLoading": false,
  "number": 1,
}
`;

exports[`features > random > RandomReducer updates state according to dispatched action 2`] = `
{
  "hasError": false,
  "isFulfilled": false,
  "isLoading": true,
  "number": undefined,
}
`;

exports[`features > random > RandomReducer updates state according to dispatched action 3`] = `
{
  "hasError": true,
  "isFulfilled": false,
  "isLoading": false,
  "number": undefined,
}
`;
